test/test_closure_pte.cpp

```diff
@@ -209,9 +229,18 @@ SCENARIO("Density-Temperature PTE Solver", "[PTE]") {
                                           copper_eos.GetOnDevice()};
       THEN("The PTE solver should converge") {
         EOS *v_EOS = copy_eos_arr_to_device(num_pte, eos_arr);
-        const bool pte_converged =
-            run_PTE_from_state(num_pte, v_EOS, spvol_bulk, sie_bulk, mass_frac);
+        Real u_bulk_out = std::numeric_limits<Real>::max();
+        const bool pte_converged = run_PTE_from_state(num_pte, v_EOS, spvol_bulk,
+                                                      sie_bulk, mass_frac, u_bulk_out);
+        // // Check that PTE converged
         // CHECK(pte_converged);
+        // // Check that PTE obeys the bulk internal energy constraint
+        // // NOTE(@pdmullen): The following fails prior to PR401
+        // const Real u_bulk = ratio(sie_bulk, spvol_bulk);
+        // const Real u_scale = std::abs(u_bulk);
+        // const Real u_bulk_scale = ratio(u_bulk, u_scale);
+        // const Real residual = std::abs(u_bulk_scale - ratio(u_bulk_out, u_scale));
+        // CHECK(residual < pte_rel_tolerance_e);
```

Comment:
This looks good (assuming you'll uncomment it). Can you put this in an `AND_THEN` clause since it's separate from convergence?

Reply:
I was planning on keeping this one commented... Notice
```c++
// Check(pte_converged)
```
itself is commented out for this test... with the accompanying comment: `// TODO: make this test converge`.  In other words, it seems that `GIVEN("A difficult two-material state")` is WIP...

The new, _active_ test is then only being applied here: https://github.com/lanl/singularity-eos/pull/401/files#diff-c129bbafa2a818eacbd83e80aa4797afe3a8d4ef822276feaa250a77eada78fcR208-R214 ; i.e., for the ` GIVEN("A difficult three-material state")` test.

I'll try to add the `AND_THEN` clause.

Reply:
Yep! That is correct. I am intending this test as a repository of states that fail even if we don't _yet_ have fixes. That way people can document bad behavior without the need to necessarily fix it yet.

I missed your test above. Sorry about that. Yes please just put that in an `AND_THEN` clause. You can simply move your comment into the label string for the `AND_THEN` clause.

From a testing perspective, it just helps separate out the checks to ensure that we understand exactly _what_ is being tested.

Reply:
Added the `AND_THEN` clause in [86232de](https://github.com/lanl/singularity-eos/pull/401/commits/86232de2c1153a7d8f4e7e99b192524826468a87)

Reply:
Looks good